azure-iot-device/tests/common/pipeline/test_operation_flow.py

```diff
@@ -63,58 +58,48 @@ def test_completes_original_op_after_new_op_completes(self, stage, op, new_op, c
         op.callback = callback
         new_op.action = "pend"
 
-        delegate_to_different_op(stage, original_op=op, new_op=new_op)
+        stage._send_worker_op_down(worker_op=new_op, op=op)
         assert callback.call_count == 0  # because new_op is pending
 
-        complete_op(stage.next, new_op)
+        stage.next._complete_op(new_op)
         assert_callback_succeeded(op=op)
 
     @pytest.mark.it("Returns the new op failure in the original op if new op fails")
     def test_returns_new_op_failure_in_original_op(self, stage, op, new_op, callback):
         op.callback = callback
         new_op.action = "fail"
-        delegate_to_different_op(stage, original_op=op, new_op=new_op)
-        assert_callback_failed(op=op, error=new_op.error)
+        stage._send_worker_op_down(worker_op=new_op, op=op)
+        assert_callback_failed(op)
 
 
-@pytest.mark.describe("pass_op_to_next_stage()")
+@pytest.mark.describe("_send_op_down()")
 class TestContinueOp(object):
-    @pytest.mark.it("Completes the op without continuing if the op has an error")
-    def test_completes_op_with_error(self, mocker, stage, op, arbitrary_exception, callback):
-        op.error = arbitrary_exception
-        op.callback = callback
-        pass_op_to_next_stage(stage, op)
-        assert_callback_failed(op=op, error=arbitrary_exception)
-        assert stage.next.run_op.call_count == 0
-
     @pytest.mark.it("Fails the op if there is no next stage")
     def test_fails_op_when_no_next_stage(self, stage, op, callback):
         op.callback = callback
         stage.next = None
-        pass_op_to_next_stage(stage, op)
+        stage._send_op_down(op)
         assert_callback_failed(op=op)
```

Comment:
This should also assert the correct type of error (in this case `PipelineError`)